File: trip-kit-booking/src/main/java/com/skedgo/tripkit/booking/BookingApi.java
package com.skedgo.tripkit.booking;

import io.reactivex.Observable;
import retrofit2.Response;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Url;

public interface BookingApi {
    @GET
    Observable<Response<BookingForm>> getFormAsync(@Url String url);

    @POST
    Observable<Response<BookingForm>> postFormAsync(
        @Url String url,
        @Body InputForm inputForm
    );

    @POST
    Observable<Response<BookingForm>> postFormAsync(@Url String url);

    @POST
    Observable<Response<BookingForm>> postFormAsync(
        @Url String url,
        @Body ActionInputForm inputForm
    );

}

File: trip-kit-booking/src/main/java/com/skedgo/tripkit/booking/BookingApi.kt
package com.skedgo.tripkit.booking

import io.reactivex.Observable
import retrofit2.Response
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Url

interface BookingApi {
    @GET
    fun getFormAsync(@Url url: String): Observable<Response<BookingForm>>

    @POST
    fun postFormAsync(
        @Url url: String,
        @Body inputForm: InputForm
    ): Observable<Response<BookingForm>>

    @POST
    fun postFormAsync(@Url url: String): Observable<Response<BookingForm>>

    @POST
    fun postFormAsync(
        @Url url: String,
        @Body inputForm: ActionInputForm
    ): Observable<Response<BookingForm>>
}